internal/controller/messaging/subscriber.go

```diff
@@ -0,0 +1,178 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package messaging
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/responses"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
+
+	"github.com/edgexfoundry/device-sdk-go/v2/internal/application"
+	"github.com/edgexfoundry/device-sdk-go/v2/internal/container"
+)
+
+type key string
+
+const (
+	correlationHeaderKey       key = common.CorrelationHeader
```

Comment:
this solved linter error: https://staticcheck.io/docs/checks#SA1029
reference: https://go.dev/blog/context#package-userip

Reply:
@lenny-intel I just realized you discussed this check in https://github.com/edgexfoundry/edgex-go/pull/3735#discussion_r718808935
Should we revert this change ?

This is causing bug in https://github.com/edgexfoundry/device-sdk-go/blob/32167e186cd280efad328d3bc9a5f43c05eae3ff/internal/application/command.go#L69

We're passing `string` type key instead of `coorelationHeaderKey` type key to `FromContext` function:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/1091de38da4f9386e49ea8f5d1470a6e592df9b2/clients/http/utils/common.go#L29

Reply:
Dang it, I should have caught that, :-(

Yes, we need to revert. Put a TODO to fix properly in 3.0